File: app/src/main/java/com/funnythingz/register/model/AppCoinFactory.java
package com.funnythingz.register.model;

import com.google.gson.Gson;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

/**
 */
public class AppCoinFactory {

    ArrayList<String> mResponseList;

    public AppCoinFactory(ArrayList<String> responseList) {
        mResponseList = responseList;
    }

    public ArrayList<AppCoin> createProductCoinList() throws JSONException {

        ArrayList<AppCoin> appCoinList = new ArrayList<AppCoin>();

        InAppBillingProduct inAppBillingProduct;
        for (String response : mResponseList) {
            JSONObject object = new JSONObject(response);
            Gson gson = new Gson();
            inAppBillingProduct = gson.fromJson(String.valueOf(object), InAppBillingProduct.class);
            appCoinList.add(
                    new AppCoin(
                            new ProductId(inAppBillingProduct.getProductId()),
                            inAppBillingProduct.getTitle(),
                            inAppBillingProduct.getPrice()
                    )
            );
        }

        return appCoinList;
    }
}

File: app/src/main/java/com/funnythingz/register/model/AppCoinFactory.kt
package com.funnythingz.register.model

import com.google.gson.Gson

import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

class AppCoinFactory(internal val mResponseList: ArrayList<String>) {

    @Throws(JSONException::class)
    fun createProductCoinList(): ArrayList<AppCoin> {

        val appCoinList = ArrayList<AppCoin>()

        for (response in mResponseList) {
            val jsonObject = JSONObject(response)
            val gson = Gson()
            val inAppBillingProduct: InAppBillingProduct = gson.fromJson(jsonObject.toString(), InAppBillingProduct::class.java)
            appCoinList.add(
                    AppCoin(ProductId(inAppBillingProduct.productId),
                            inAppBillingProduct.title,
                            inAppBillingProduct.price))
        }

        return appCoinList
    }
}